src/item_factory.cpp

```diff
@@ -2608,31 +2608,97 @@ void Item_factory::add_special_pockets( itype &def )
     }
 }
 
-static int parse_acc_string( const std::string &to_hit )
+namespace acc_parse
+{
+// *INDENT-OFF*
+enum class grip_val : int {
+    BAD = -1,
+    NONE,
+    SOLID,
+    WEAPON
+};
+enum class length_val : int {
+    HAND = 0,
+    SHORT,
+    LONG
+};
+enum class surface_val : int {
+    POINT = -2,
+    LINE,
+    ANY,
+    EVERY
+};
+enum class balance_val : int {
+    CLUMSY = -2,
+    UNEVEN,
+    NEUTRAL,
+    GOOD
+};
```

Comment:
These enums need a final member (NUM), and this: 
```c++
template<>
struct enum_traits<ENUM> {
       static constexpr ENUM last = ENUM::NUM;
};
```
to be used for enum_to_string.